sympy/printing/tests/test_ccode.py

```diff
@@ -217,8 +217,11 @@ def test_ccode_Indexed_without_looking_for_contraction():
     Dy = IndexedBase('Dy', shape=(len_y-1,))
     i = Idx('i', len_y-1)
     e=Eq(Dy[i], (y[i+1]-y[i])/(x[i+1]-x[i]))
-    code0 = ccode(e.rhs, assign_to=e.lhs, contract=False)
-    assert code0 == 'Dy[i] = (y[%s] - y[i])/(x[%s] - x[i]);' % (i + 1, i + 1)
+    code0 = ccode(e.rhs, assign_to=e.lhs)
+    expected = ("for (int i=0; i<4; i++){\n"
+                "   Dy[i] = (y[i + 1] - y[i])/(x[i + 1] - x[i]);\n"
+                "}")
+    assert code0 == expected
 
```

Comment:
There needs to be a test for generating a "naked" assignment (not wrapped in a loop).